app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -344,15 +344,26 @@ export default Component.extend({
     return toolbar;
   },
 
+  cachedCookAsync(text) {
```

Comment:
You could probably make it even simpler by setting `this.cookFunction` in an `init()` method when the component is created and then you could call it without checking if it exists every time. I doubt it would change performance much but it would be simpler code.

Reply:
The problem is that loading pretty-text is asynchronous. So we can't set it synchronously during `init`. 

I guess we could trigger the loadScript during `init`, but then we would still have to check whether loading is done before cooking anything.